src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -253,19 +255,54 @@ ss::future<bool> sharded_store::upsert(
   schema_id id,
   schema_version version,
   is_deleted deleted) {
-    try {
-        schema = co_await make_canonical_schema(
-          schema.share(), normalize::no, false);
-    } catch (...) {
-        // do nothing. In case make_canonical_schema fails, continue with the
-        // schema as-is in the topic
+    bool should_mark = false;
```

Comment:
nitpick: I'd have been tempted to use `canonical_fut.failed()` as the argument, or at least initialise `bool should_mark = canonical_fut.failed()`